### ISSUE TITLE
Codemaid makes Visual Studio crash/restart when reorganizing a document of around 17.000 lines.

### ISSUE THREAD

**Issue by ElektroStudios** [codecadwallader/codemaid#655]
### Environment
- Visual Studio version: 2017 Pro
- CodeMaid version: 11.0
- Code language: VB.NET

### Description
Codemaid makes Visual Studio crash/restart when reorganizing a document of around 17.000 lines.

### Steps to recreate
1. Just try to reorganize this code using CodeMaid:
    - https://paste.ubuntu.com/p/CWQQHr6fbY/

**Comment by codecadwallader** [comment]
Thank you for reporting the issue.  I am able to reproduce it, and this is a known issue that the larger the file the larger the resources required to process it.  As you've seen breaking up the file is the known workaround.  There is some more information on #83 if it helps.

**Comment by ElektroStudios** [comment]
Thanks for comment... after 13 days of the issue report publication (alert superstitious people), but could you please extend your answer to clarify whether or not this issue could be inspected and finally fixed directly in **CodeMaid**?. Splitting the original file into smaller total amount of lines is not a viable option in my scenario.

In the thread that you linked, you stated that "_enabling multi-threading performance (which is off by default) actually makes things worse for some users_", however I can't find any multi-threading related option in **CodeMaid** settings to try disabling/enabling it and see the results.

After read that thread, I don't know whether **CodeMaid** still rely on **VsRegexFinder** class or on **Roslyn**. But the thread is from year 2015 and there you stated "_In the future (VS2015+) the COM APIs go away and CodeMaid can utilize Roslyn which may bypass the issue._" and that gives me some hope about being able to fixinf this issue...

In short, could you do/think something to fix the issue in **CodeMaid**?.

Thanks for your time and support.

**Comment by codecadwallader** [comment]
We are still heavily relying on COM APIs for CodeMaid so those memory issues linked in the other issue are still applicable.  We still plan to move over to Roslyn but it's a significant effort and a long running effort.  Roslyn isn't a magic bullet either to be candid, the larger the file the larger the resource demands.

We took away multi-threading performance in a later version, so that thread comment is no longer applicable.

**Comment by StevenTCramer** [comment]
I don't mean to be flippant. 17000 lines, in a single file?  Is there a reason you need that big of a file?  Can you not at least split it into 2 files, if not a 100?

**Comment by ElektroStudios** [comment]
My project provides a vast amount of P/Invokes from Win32 libraries. For example I have a file with name **User32.vb**, which provides functions from User32.dll. Every function in my source-code is xml documented, and this increases the total amount of lines in the file, a lot. A similar thing would be [Vanara's source-code](https://github.com/dahall/Vanara/tree/master/PInvoke). 

Is not necessary to have it all in a file of 17.000 lines, of course I could split them into multiple files like for example you can see in Vanara's source code (which anyway have files of more than 9.000 lines with P/Invoke definitions), but having all the P/invokes from one dll file in a single class file, would be the way to keep it ordered by name... if I could use a tool like CodeMaid to sort function members by name in that file; and after that maybe I could start splitting them into two files from A-M and N-Z or by thematic, but to do that first I need to be able sort the members name, and I can't do that due CodeMaid can't handle that amount of lines, and I can't do that task manually without automation...

PS: please note that online C# source-code reference from Microsoft Team has class files of much more than 20.000 lines. Sometimes it is the way to do things, not a bad habit.

**Comment by StevenTCramer** [comment]
Ok thanks for the explanation.

In the mean time you could split the file before sorting, into the size CM could handle, and then sort. Then move the top of each to appropriate file and resort. Manual I know.

Or I guess digin and try a PR, not easy either.

Thanks again, I was just curious what kind of code would have 17k lines per file.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by akeijzer11** [comment]
> I was just curious what kind of code would have 17k lines per file

How do you manage code changes in periodically auto-generated files?

In my scenario, the code for the (connected service) WCF Web Service is auto-generated from WSDLs. Visual Studio employs svcutil to generate C# code in a single file without any specific order (apart from the WSDL's structure). This lack of order makes it challenging to track changes when updating the file with a new (or even the same) WSDL, due to the disorder among the file's classes.

To tackle this issue, I've employed CodeMaid in Visual Studio 2022 to alphabetically reorganize the classes in the generated file based on class names.

The current reorganization process has been running for over 24 hours on my i9-12900 with 64GB DDR5 RAM and a PCIE SSD.
![image](https://github.com/codecadwallader/codemaid/assets/25773759/66a76706-b368-4baf-a6c6-ef2e508fb865)

While the file appears to be organized (with visible changes in order), attempting to save it results in Visual Studio crashing. It seems that CodeMaid has made some progress, but I'm uncertain about the completion status.

I'm curious if there's an alternative approach aside from splitting the file into multiple parts. Unfortunately, my files exceeds 100k lines, making splitting impractical for me. Any suggestions or insights?
